### PR TITLE
source-build: support building aspnetcore using non-portable runtime packages.

### PR DESCRIPTION
**Pull request by tmds**
Currently source-build performs 'runtime-portable' build that produces 'linux-{arch}' packages that are used when building ASP.NET Core.

With this change, we can use the non-portable packages that are produced by the source-build 'runtime' build, and eliminate the 'runtime-portable' build.

cc @omajid @MichaelSimons @crummel

### CODE REVIEW

File: eng/Common.props
Comment (tmds) [reply eng/Common.props:10]: I'll move it to `Directory.Build.props`.

Comment (dougbu) [reply eng/Common.props:10]: OIC this is fine, sorry. thought you were changing a file under eng/common/

File: eng/Dependencies.props
Comment (tmds) [reply eng/Dependencies.props:109]: If we're doing a portable build (like `linux-x64`), we'll add the `LastPackageReference` for the rid twice and this gives `NETSDK1023` in `Microsoft.AspNetCore.App.Runtime.csproj`.

I don't know why it gives that error if include it twice, but not once.

The condition is here to only add `LatestPackageReference` for `TargetRuntimeIdentifier` when it isn't portable (like `fedora.36-x64`).

File: src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj
Comment (tmds) [reply src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj:99]: `fedora.36-x64` for example.

File: src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj
Comment (tmds) [reply src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj:108]: I'll revert to the previous name.

File: src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj
Comment (tmds) [reply src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj:111]: `fedora.36-x64` for example.

**Comment by dougbu** [comment]
The CI build is just hitting a known area of flaky tests. However, that doesn't validate the `'$(PortableBuild)' == 'false'` case. Have you tested the new scenario somewhere❔

**Comment by dougbu** [comment]
@MichaelSimons what are your thoughts on this PR and how to test it prior to merging❔ I think it should do what it says on the tin but am not confident.

**Comment by tmds** [comment]
> @MichaelSimons what are your thoughts on this PR and how to test it prior to merginggrey_question I think it should do what it says on the tin but am not confident.

I created a PR in installer to help validate this change: https://github.com/dotnet/installer/pull/14549.
And https://github.com/dotnet/source-build/issues/3027 describes why the change is being made.

**Comment by dougbu** [comment]
> I created a PR in installer to help validate this change: dotnet/installer#14549.

Should we get this in now or wait for that PR to be merged❔

**Comment by MichaelSimons** [comment]
> Should we get this in now or wait for that PR to be merged❔

I am in favor of merging now.

**Comment by dougbu** [comment]
Enabled auto-merge. New CI build should pick up some reliability fixes from our main branch and get this in today.

File: eng/Common.props
Comment (ViktorHofer) [inline eng/Common.props:10]: Unsure what the existing pattern is but usually in the msbuild world when you want to mimic a boolean, you condition on something either being true or not being true. With conditions that assert on `true` and `false` you are allowing the property to have values that aren't covered by these conditions.

TL;DR: I would expect this to be `'$(PortableBuild)' != 'true'`

Comment (tmds) [reply eng/Common.props:10]: I had first written it like that, and then inverted it in https://github.com/dotnet/aspnetcore/pull/43937/commits/82414df0502229bc5685e048870f7309729af0ef because of the build failures: https://github.com/dotnet/aspnetcore/runs/8326525459.

I rewrote the check that in case `PortableBuild` is not set, the checks default to portable behavior.

**Comment by ViktorHofer** [comment]
@dougbu Auto merge doesn't seem to work for this PR

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
Just needed an approval 😁

**Comment by tmds** [comment]
@dougbu can you initiate the backport to 7.0 for this? It has already been backported to 6.0 in https://github.com/dotnet/aspnetcore/pull/44752?

**Pull request merged**
